src/prefect/cli/run.py

```diff
@@ -1,42 +1,688 @@
+        "multiple labels. If not passed, the labels from the flow group will be used."
+    ),
+    multiple=True,
+)
+@click.option("--run-name", help="A name to assign to the flow run.", default=None)
+@click.option(
+    "--context",
+    "context_vars",
+    help=(
+        "A key, value pair (key=value) specifying a flow context variable. The value "
+        "will be interpreted as JSON. May be passed multiple times to specify multiple "
+        "context values. Nested values may be set by passing a dict."
+    ),
+    multiple=True,
+)
+@click.option(
+    "--param",
+    "params",
+    help=(
+        "A key, value pair (key=value) specifying a flow parameter. The value will be "
+        "interpreted as JSON. May be passed multiple times to specify multiple "
+        "parameter values."
+    ),
+    multiple=True,
+)
+@click.option(
+    "--log-level",
+    help=(
+        "The log level to set for the flow run. "
+        "If not set, the default level will be used."
+    ),
+    type=click.Choice(
+        ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"], case_sensitive=False
+    ),
+    default=None,
+)
+@click.option(
+    "--param-file",
+    help=(
+        "The path to a JSON file containing parameter keys and values. Any parameters "
+        "passed with `--param` will take precedence over these values."
+    ),
+    default=None,
+)
+# Display settings ---------------------------------------------------------------------
+@click.option(
+    "--quiet",
+    "-q",
+    help=(
+        "Disable verbose messaging about the flow run and just print the flow run id."
+    ),
+    is_flag=True,
+)
+@click.option(
+    "--no-logs",
+    help=(
+        "Disable streaming logs from the flow run to this terminal. Only state changes "
+        "will be displayed. Only applicable when `--watch` is set."
+    ),
+    is_flag=True,
+)
+@click.option(
+    "--watch",
+    "-w",
+    help="Wait for the flow run to finish executing and display status information.",
+    is_flag=True,
+)
+@handle_terminal_error
+def run(
+    ctx,
+    flow_or_group_id,
+    project,
+    path,
+    module,
+    name,
+    labels,
+    context_vars,
+    params,
+    log_level,
+    param_file,
+    run_name,
+    quiet,
+    no_logs,
+    watch,
+):
+    """Run a flow"""
+    # Since the old command was a subcommand of this, we have to do some
+    # mucking to smoothly deprecate it. Can be removed with `prefect run flow`
+    # is removed.
+    if ctx.invoked_subcommand is not None:
+        if any([params, no_logs, quiet, flow_or_group_id]):
+            # These options are not supported by `prefect run flow`
+            raise ClickException(
+                "Got unexpected extra argument (%s)" % ctx.invoked_subcommand
+            )
+        return
+
+    # Define a simple function so we don't have to have a lot of `if not quiet` logic
+    def quiet_echo(*args, **kwargs):
+        if not quiet:
```

Comment:
Slight preference for not relying on `quiet` within `quiet_echo` here, could be a pain to track down failures if `quiet` is modified

something like this maybe? idk if there's a more compact way to express it
```python
if quiet:
    def quiet_echo(*args, **kwargs):
        pass
else:
    def quiet_echo(*args, **kwargs):
        click.secho(*args, **kwargs)
```

Reply:
Ahh ehh hmm that's a fair point. Seems like it'd be really weird to mutate `quiet` but your solution seems safer